test/phoenix/integration/socket_driver_test.exs

```diff
@@ -0,0 +1,301 @@
+defmodule Phoenix.Integration.SocketDriverTest do
```

Comment:
In Phoenix suite, we call integration tests the ones that go through an actual client like HTTP or Websocket. This is a regular test, so I would place it under `phoenix/socket/driver_test.exs`.